warp_repulsion: 0.0001